### ISSUE TITLE
Inquiry about Astroid Framework Version 3.0.7 Compatibility with PHP 8

### ISSUE THREAD

**Issue by salahjaafar** [templaza/astroid-framework#501]
Hello Astroid Team,

I hope you are doing well. I am currently using Astroid Framework Version 3.0.7 in a Joomla project with php 7.4, and I was wondering if this version is only compatible with PHP 8.

Furthermore, I have encountered the following error while using the administrative part:

```
Fatal error: Default value for property of type Astroid\mixed may not be null. Use the nullable type ?Astroid\mixed to allow null default value in /opt/lampp/htdocs/joomla/libraries/astroid/framework/library/astroid/Element.php on line 18
```

Is this error related to an incompatibility with PHP 8, or could it be related to something else?

Thank you in advance for your assistance and for the fantastic work you are doing with Astroid Framework!

Best regards,
Salah JAAFAR

**Comment by salahjaafar** [comment]
Yes, it depends on migrating all PHP projects to work on the same server.

**Comment by sonvnn** [comment]
Php7 is end of life. You have to update it first before update your Joomla. Joomla 5 is not support Php7 even php 8.0 will be drop support in  next year. You should prepare for it.

I recommend you delete all of extensions which did not update to php 8.1. Because they are not safe. Your website will be hacked because one of them.

Best regards,
Sonny

**Comment by salahjaafar** [comment]
Certainly, it is imperative to engage in the migration process. While this may pose some challenges given the simultaneity of several ongoing projects, this undertaking will become an unavoidable necessity. @sonvnn i would like to express my gratitude for all your efforts.

**Comment by felixkat** [comment]
Closing due to solution provided.
